xdsl/ir.py

```diff
@@ -1102,16 +1103,26 @@ def __hash__(self) -> int:
         return id(self)
 
 
-@dataclass
+@dataclass(init=False)
 class Region(IRNode):
     """A region contains a CFG of blocks. Regions are contained in operations."""
 
-    blocks: list[Block] = field(default_factory=list, init=False)
+    blocks: list[Block]
     """Blocks contained in the region. The first block is the entry block."""
 
-    parent: Operation | None = field(default=None, init=False, repr=False)
+    parent: Operation | None = field(repr=False)
     """Operation containing the region."""
 
+    def __init__(self,
+                 blocks: Iterable[Block] = (),
+                 *,
```

Comment:
What does this do exactly? :thinking:

Reply:
It means that the arguments after this one are name-only, so the user can't go `Region(x, y)`, they must spell it `Region(x, parent=y)`